### ISSUE TITLE
gcc-12 fails to build from cache on Ubuntu 23.10 (Mantic)

### ISSUE THREAD

**Issue by rbalint** [firebuild/firebuild#1267]
Observed with: Firebuild Git v0.8.2-122-gdf2ac5b

```
checking for ttyname_r... configure: error: unsupported system, cannot find Fortran integer kind for omp_depend_kind
no
checking for clock_gettime... make[3]: *** [Makefile:16582: configure-target-libgomp] Error 1
make[3]: *** Waiting for unfinished jobs....
no
```

**Comment by rbalint** [comment]
gcc-13 fails in a similar way:

```
checking for chsize... no
checking for chdir... configure: error: unsupported system, cannot find Fortran int kind=16, needed for o
mp_depend_kind
no
checking for getentropy... make[3]: *** [Makefile:16397: configure-target-libgomp] Error 1
make[3]: *** Waiting for unfinished jobs....
no
checking for getlogin... no
```

**Comment by rbalint** [comment]
This seems to be resolved by #1281 as a different manifestation of #1268 .
